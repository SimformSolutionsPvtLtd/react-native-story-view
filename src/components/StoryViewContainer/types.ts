import type { ViewToken } from 'react-native';
import type { SharedValue } from 'react-native-reanimated';
import type { GestureHandlerEvent } from 'react-native-reanimated/lib/typescript/reanimated2/hook';
import type { StoriesType, StoryContainerProps } from '../StoryView/types';

export enum TransitionMode {
  Default,
  Cube,
  Scale,
}

export interface StoryViewContainerProps
  extends Omit<StoryContainerProps, 'stories'> {
  stories: StoriesType[];
  onComplete?: () => void;
  onUserStoryIndexChange?: (index: number) => void;
  userStoryIndex?: number;
  visible?: boolean;
  viewedStories: Array<boolean[]>;
  onChangePosition?: (
    storyIndex: number,
    userIndex?: number
  ) => void | undefined;
  transitionMode?: TransitionMode;
}

export interface StoryViewListItemProps
  extends Omit<StoryContainerProps, 'stories'> {
  item: StoriesType;
  index: number;
  storyIndex: number;
  viewedStories: Array<boolean[]>;
  nextStory?: () => void;
  previousStory?: () => void;
  onComplete?: () => void;
  transitionMode?: TransitionMode;
  scrollX: ScrollValue;
  isTransitionActive: boolean;
  gestureHandler?: (e: GestureHandlerEvent<any>) => void;
}

export interface ViewConfig {
  viewableItems: Array<ViewToken>;
  changed: Array<ViewToken>;
}

export interface ListItemProps {
  item: StoriesType;
  index: number;
}

export interface ListItemRef {
  onScrollBegin: () => void;
  onScrollEnd: () => void;
  handleLongPress: (visibility: boolean) => void;
}

export interface DraggableGestureProps {
  backgroundColor?: string;
  onComplete?: () => void;
  onScrollBeginDrag: () => void;
  onScrollEndDrag: () => void;
  handleLongPress: (visibility: boolean) => void;
  isKeyboardVisible: boolean;
}

export type ScrollValue = SharedValue<number>;
