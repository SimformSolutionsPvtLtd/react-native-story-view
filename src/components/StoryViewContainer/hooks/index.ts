import useStoryViewContainer from './useStoryViewContainer';

export { useStoryViewContainer };
