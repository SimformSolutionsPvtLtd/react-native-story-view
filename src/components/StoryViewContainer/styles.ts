import { StyleSheet } from 'react-native';
import { Colors, Metrics, moderateScale } from '../../theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: moderateScale(40),
    backgroundColor: Colors.black,
  },
  itemContainer: {
    flex: 1,
    width: Metrics.screenWidth,
  },
  touchContainer: {
    flex: 1,
  },
  list: {
    flex: 1,
    backgroundColor: Colors.black,
  },
  loaderStyle: {
    flex: 1,
    alignSelf: 'center',
  },
});

export default styles;
