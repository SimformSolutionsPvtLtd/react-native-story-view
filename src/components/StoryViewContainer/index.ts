import StoryViewContainer from './StoryViewContainer';

export { TransitionMode } from './types';
export type { StoryViewContainerProps } from './types';
export { StoryViewContainer };
