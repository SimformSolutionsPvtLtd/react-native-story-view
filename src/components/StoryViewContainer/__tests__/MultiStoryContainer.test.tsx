import React from 'react';
import { render } from '@testing-library/react-native';
import StoryViewContainer from '../StoryViewContainer';

jest.useFakeTimers();

describe('StoryViewContainer Component', () => {
  it('Match Snapshot', () => {
    const { toJSON } = render(
      <StoryViewContainer visible={false} stories={[]} viewedStories={[]} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
