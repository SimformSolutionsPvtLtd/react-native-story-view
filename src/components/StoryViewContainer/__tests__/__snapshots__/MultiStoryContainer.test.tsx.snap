// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryViewContainer Component Match Snapshot 1`] = `null`;
